From d441349df5b25bbb8afe93028d36aedac0b46550 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Mon, 20 Feb 2017 12:37:28 -0500
Subject: [PATCH 1/3] ARM: dts: k2g-ice: add tx-internal-delay for dp83867
 clock path

EVM hardware spec recommends to add 0.25 nsec delay in the tx
direction for internal delay in the clock path to be on the safer
side. Also change the phy-mode to RGMII-ID to apply both tx and rx
delays.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 arch/arm/dts/k2g-ice.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/k2g-ice.dts b/arch/arm/dts/k2g-ice.dts
index 76cfad0..4224439 100644
--- a/arch/arm/dts/k2g-ice.dts
+++ b/arch/arm/dts/k2g-ice.dts
@@ -82,6 +82,7 @@
 	ethphy0: ethernet-phy@0 {
 		reg = <0>;
 		ti,rx-internal-delay = <DP83867_RGMIIDCTL_2_25_NS>;
+		ti,tx-internal-delay = <DP83867_RGMIIDCTL_250_PS>;
 		ti,fifo-depth = <DP83867_PHYCR_FIFO_DEPTH_8_B_NIB>;
 		ti,min-output-impedance;
 	};
@@ -89,5 +90,5 @@
 
 &gbe0 {
 	phy-handle = <&ethphy0>;
-	phy-mode = "rgmii-rxid";
+	phy-mode = "rgmii-id";
 };
-- 
1.9.1

